System.IO.FileSystem/File.cs

```diff
@@ -88,64 +122,95 @@ public static FileStream Create(string path)
         /// <exception cref="IOException">Directory is not found or <paramref name="path"/> is read-only or a directory.</exception>
         public static void Delete(string path)
         {
-            if (string.IsNullOrEmpty(path))
-            {
-                throw new ArgumentException();
-            }
+            // path validation happening in the call
+            path = Path.GetFullPath(path);
+            string folderPath = Path.GetDirectoryName(path);
+
+            // make sure no one else has the file opened, and no one else can modify it when we're deleting
+            object record = FileSystemManager.AddToOpenList(path);
 
             try
             {
-                byte attributes;
-                var directoryName = Path.GetDirectoryName(path);
+                uint attributes = NativeIO.GetAttributes(folderPath);
 
-                // Only check folder if its not the Root
-                if (directoryName != Path.GetPathRoot(path))
+                // in case the folder does not exist or is invalid we throw DirNotFound Exception
+                if (attributes == 0xFFFFFFFF)

```

Comment:
just as a remqrk and because it appears in many places: should the 0xFFFFFFFF constant be in the FileManager and just used here as a constant? If that changes for some reasons, at least it has to change only in 1 single place. Now it's over multiple files and in few places in those files.

Reply:
Done!